org/mozilla/jss/util/JDKCompat.java

```diff
@@ -0,0 +1,21 @@
+package org.mozilla.jss.util;
+
+import java.lang.reflect.Method;
+
+import javax.net.ssl.SSLParameters;
+
+public class JDKCompat {
+    public static class SSLParametersHelper {
+        public static String[] getApplicationProtocols(SSLParameters inst) {
+            try {
+                Method getter = inst.getClass().getMethod("getApplicationProtocols");
+                Object result = getter.invoke(inst);
+                if (result instanceof String[]) {
+                    return (String[]) result;
+                }
+            } catch (Throwable t) {}
```

Comment:
If we want to make ALPN support optional, we should only catch `NoSuchMethodException` and explicitly ignore or a generate warning:
```
} catch (NoSuchMethodException e) {
    // ignore
}
```
All other exceptions should still be propagated to the callers.

Reply:
Also, it's not necessary to check the type of `result` since it's always `String[]` in all Java versions (if the method is available).